src/daria/analysis/segmentationcomparison.py

```diff
@@ -0,0 +1,118 @@
+    )
+    gray_base: np.ndarray = np.fix(gray_colors / N).astype(np.uint8)
+
+    # Define unique colors
+    colors: np.ndarray = kwargs.pop(
+        "colors",
+        np.array(
+            [
+                [[205, 0, 0]],
+                [[0, 205, 0]],
+                [[0, 0, 205]],
+                [[205, 205, 0]],
+                [[205, 0, 205]],
+                [[0, 205, 205]],
+            ],
+            dtype=np.uint8,
+        ),
+    )
+    light_scaling: float = kwargs.pop("light_scaling", 1.5)
+    colors_light: np.ndarray = np.trunc(light_scaling * colors)
+    np.clip(colors_light, 0, 255, out=colors_light)
+    colors: np.ndarray = np.hstack((colors, colors_light))
+
+    # Assert that there are a sufficient amount of colors
+    # and that all of the segmentations are of equal size
+    assert len(colors[:, 0, 0]) >= N
+    for i in range(N - 1):
+        assert segmentations[0].shape == segmentations[i + 1].shape
+
+    # Create the return array
+    return_image: np.ndarray = np.full(
+        shape=segmentations[0].shape + (3,), fill_value=[0, 0, 0], dtype=np.uint8
+    )
+
+    # Enter gray everywhere there are ovelaps of different segmentations
+    for k in range(N):
+        for i in range(k + 1, N):
+
+            # Overlap of first component
+            return_image[
+                np.where(
+                    np.logical_and(
+                        segmentations[k] == components[0],
+                        segmentations[i] == components[0],
+                    )
+                )
+            ] += gray_base[0]
+
+            # Overlap of second component
+            return_image[
+                np.where(
+                    np.logical_and(
+                        segmentations[k] == components[1],
+                        segmentations[i] == components[1],
+                    )
+                )
+            ] += gray_base[1]
+
+            # Overlap of different components
+            return_image[
+                np.where(
+                    np.logical_or(
+                        np.logical_and(
+                            segmentations[k] == components[0],
+                            segmentations[i] == components[1],
+                        ),
+                        np.logical_and(
+                            segmentations[k] == components[1],
+                            segmentations[i] == components[0],
+                        ),
+                    )
+                )
+            ] += gray_base[2]
+
+    # Determine locations (and make modifications to return image) of unique coomponents
+    for c in components:
+        for k in range(N):
+            only_tmp: np.ndarray = segmentations[k] == c
+            for j in filter(lambda j: j != k, range(N)):
+                only_tmp = np.logical_and(only_tmp, segmentations[j] == 0)
+
+            return_image[np.where(only_tmp)] = colors[k, components.index(c)]
```

Comment:
Try without np.where here as well. And I would usually use `enumerate` here instead of `index`. Since the number of components won't be large, the cost will be bounded here. So it is not critical, but I think it would be the more standard way of doing this.